project/infra/code-index/src/arcadedb_client.py

```
@@ -0,0 +1,312 @@
+            if where_conditions:
+                where_clause = "WHERE " + " AND ".join(where_conditions)
+
+            # Execute vector search query
+            embedding_json = json.dumps(embedding)
+            query = f"""
+                SELECT *, vectorDistance(embedding, {embedding_json}) as score
+                FROM CodeChunk
+                {where_clause}
+                ORDER BY score ASC
+                LIMIT {top_k}
+            """
```

Ruff (0.15.2)

[error] Possible SQL injection vector through string-based query construction

(S608)